backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/ddl/public/functions/execute_surcharge_calculation_SQL.sql

```diff
@@ -0,0 +1,35 @@
+DROP FUNCTION IF EXISTS execute_surcharge_calculation_SQL(numeric, numeric, numeric)
+;
+
+CREATE OR REPLACE FUNCTION execute_surcharge_calculation_SQL
+(
+    p_m_discountschema_id numeric,
+    p_Target_PriceList_Version_ID numeric,
+    p_Source_M_ProductPrice_ID numeric
+)
+    RETURNS numeric AS
+$BODY$
+DECLARE
+    sql text;
+    result numeric;
+BEGIN
+
+    sql := (
+        SELECT dsc.surcharge_calc_sql
+        FROM M_DiscountSchema ds
+            LEFT JOIN M_DiscountSchema_Calculated_Surcharge dsc ON ds.M_DiscountSchema_Calculated_Surcharge_ID = dsc.M_DiscountSchema_Calculated_Surcharge_ID
+        WHERE ds.m_discountschema_id = p_m_discountschema_id
+    );
+
+    if sql IS NULL then
+        RETURN 0::numeric;
+    end if;
+
+    EXECUTE sql
+        USING p_Target_PriceList_Version_ID, p_Source_M_ProductPrice_ID
```

Comment:
hmmm, is the SQL code sensible to the order of params, right?

Reply:
that might be fragile and frustrating for the "power user"....

Reply:
I added a description to the field  Description='The SQL needs to contain two parameter: "$1" ( Target_PriceList_Version_ID ) und "$2" ( Source_M_ProductPrice_ID )